### PR TITLE
Add dev.mariinkys.Oboete

### PR DESCRIPTION
**Pull request by mariinkys**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly.

A simple flashcards study application made with libcosmic.

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements#control-over-domain
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: dev.mariinkys.Oboete.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                "install -Dm0644 ./res/icons/hicolor/16x16/apps/dev.mariinkys.Oboete.svg /app/share/icons/hicolor/16x16/apps/dev.mariinkys.Oboete.svg",
+                "install -Dm0644 ./res/icons/hicolor/24x24/apps/dev.mariinkys.Oboete.svg /app/share/icons/hicolor/24x24/apps/dev.mariinkys.Oboete.svg",
+                "install -Dm0644 ./res/icons/hicolor/32x32/apps/dev.mariinkys.Oboete.svg /app/share/icons/hicolor/32x32/apps/dev.mariinkys.Oboete.svg",
+                "install -Dm0644 ./res/icons/hicolor/48x48/apps/dev.mariinkys.Oboete.svg /app/share/icons/hicolor/48x48/apps/dev.mariinkys.Oboete.svg",
+                "install -Dm0644 ./res/icons/hicolor/64x64/apps/dev.mariinkys.Oboete.svg /app/share/icons/hicolor/64x64/apps/dev.mariinkys.Oboete.svg",
+                "install -Dm0644 ./res/icons/hicolor/128x128/apps/dev.mariinkys.Oboete.svg /app/share/icons/hicolor/128x128/apps/dev.mariinkys.Oboete.svg",
+                "install -Dm0644 ./res/icons/hicolor/256x256/apps/dev.mariinkys.Oboete.svg /app/share/icons/hicolor/256x256/apps/dev.mariinkys.Oboete.svg"
```
Comment (bbhtt) [inline dev.mariinkys.Oboete.json:41]: You are installing svg icons in directories which is for PNG only

File: dev.mariinkys.Oboete.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            ],
+            "sources": [
+                {
+                    "type": "dir",
+                    "path": "."
```
Comment (bbhtt) [inline dev.mariinkys.Oboete.json:46]: How is the CI here supposed to fetch the source here without an URL?

File: dev.mariinkys.Oboete.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    "sdk-extensions": [
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command": "oboete",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--device=dri",
+        "--share=network"
```
Comment (bbhtt) [inline dev.mariinkys.Oboete.json:15]: Why do you need network? From the description it sounds like this should not be required.

File: dev.mariinkys.Oboete.json (hunk 1 above)
Comment (bbhtt) [reply dev.mariinkys.Oboete.json:41]: Also unless you need all, just install a 256x256 PNG or just a SVG.

**Comment by mariinkys** [comment]
I'm sorry, first time doing this and I'm a bit lost...

File: dev.mariinkys.Oboete.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                "install -Dm0644 ./res/icons/hicolor/256x256/apps/dev.mariinkys.Oboete.png /app/share/icons/hicolor/256x256/apps/dev.mariinkys.Oboete.png",
+                "install -Dm0644 ./res/icons/hicolor/256x256/apps/dev.mariinkys.Oboete.svg /app/share/icons/hicolor/scalable/apps/dev.mariinkys.Oboete.svg"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "path": "https://github.com/mariinkys/Oboete",
+                    "tag": "0.1",
+                    "commit": "7c684847aeb8c9af66dbcb134284ec16803a2502",
+                    "builddir": true
```
Comment (hfiguiere) [inline dev.mariinkys.Oboete.json:43]:
```suggestion
                    "commit": "7c684847aeb8c9af66dbcb134284ec16803a2502"
```
`builddir` doesn't belong here.

**Comment by bbhtt** [comment]
bot, build dev.mariinkys.Oboete

File: dev.mariinkys.Oboete.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+                "install -Dm0755 ./target/release/oboete /app/bin/oboete",
+                "install -Dm0644 ./res/dev.mariinkys.Oboete.desktop /app/share/applications/dev.mariinkys.Oboete.desktop",
+                "install -Dm0644 ./res/dev.mariinkys.Oboete.metainfo.xml /app/share/metainfo/dev.mariinkys.Oboete.metainfo.xml",
+                "install -Dm0644 ./res/icons/hicolor/256x256/apps/dev.mariinkys.Oboete.png /app/share/icons/hicolor/256x256/apps/dev.mariinkys.Oboete.png",
+                "install -Dm0644 ./res/icons/hicolor/256x256/apps/dev.mariinkys.Oboete.svg /app/share/icons/hicolor/scalable/apps/dev.mariinkys.Oboete.svg"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "path": "https://github.com/mariinkys/Oboete",
```
Comment (bbhtt) [inline dev.mariinkys.Oboete.json:40]:
```suggestion
                    "url": "https://github.com/mariinkys/oboete.git",
```

**Comment by bbhtt** [comment]
bot, build dev.mariinkys.Oboete

**Comment by mariinkys** [comment]
Tested a few times on my machine now working fine, sorry for the trouble.

**Comment by bbhtt** [comment]
bot, build dev.mariinkys.Oboete

**Comment by mariinkys** [comment]
Mmm... Just noticed asian fonts are not displaying correctly could this be due to permissions?

**Comment by mariinkys** [comment]
I've tried giving read permissions to ~/.local/share/fonts but does not seem to work

**Comment by bbhtt** [comment]
Flatpak loads fonts from system automatically. This is likely an issue on your end.

**Comment by mariinkys** [comment]
Just tested on another system running Debian 12 instead of mine and it work's fine, so yeah the issue seems to be with my Fedora 40 installation :/

**Comment by mariinkys** [comment]
Should the screenshots show up on the store when looking at the app details?

**Comment by mariinkys** [comment]
> Just tested on another system running Debian 12 instead of mine and it work's fine, so yeah the issue seems to be with my Fedora 40 installation :/

However it's interesting how this does not happen when running the app locally (not a flatpak)

**Comment by mariinkys** [comment]
After a lot of tries I succeeded on displaying Chinese and Japanese fonts on the application by copying the contents of `~/.local/share/fonts` to `~/.fonts` and giving permission to `~/.fonts` to the application with Flatseal. However it does not work if giving permission directly to `~/.local/share/fonts` and also does not make if as you said Flatpak loads fonts from system automatically.

**Comment by bbhtt** [comment]
It will load fonts automatically from `~/.local/share/fonts` but it won't work if both that and `~/.fonts` directory is present on host.

If the font needs additional fontconfig file to work properly it won't work. Flatpak won't load any fontconfig files from host.

**Comment by mariinkys** [comment]
> It will load fonts automatically from `~/.local/share/fonts` but it won't work if both that and `~/.fonts` directory is present on host.
> 
> If the font needs additional fontconfig file to work properly it won't work. Flatpak won't load any fontconfig files from host.

Well at least on my end it does not work when  `~/.fonts` does not exist and all the fonts are like normally on `~/.local/share/fonts`. Maybe I'm missing something?

**Comment by mariinkys** [comment]
> > It will load fonts automatically from `~/.local/share/fonts` but it won't work if both that and `~/.fonts` directory is present on host.
> > If the font needs additional fontconfig file to work properly it won't work. Flatpak won't load any fontconfig files from host.
> 
> Well at least on my end it does not work when `~/.fonts` does not exist and all the fonts are like normally on `~/.local/share/fonts`. Maybe I'm missing something?

Just for reference the contents of my `~/.local/share/fonts` are just three fonts: `NotoSansJP-Regular.ttf` `NotoSansKR-Regular.ttf` and `NotoSansSC-Regular.ttf`. I'm not sure what "additional fontconfig files" are but from what I could see on a google search this does not seem the case?

**Comment by bbhtt** [comment]
Delete `~/.fonts` directory, run the flatpak as `flatpak run --command=ls <appid> -lA run/host/user-fonts`, what do you see?

**Comment by mariinkys** [comment]
> Delete `~/.fonts` directory, run the flatpak as `flatpak run --command=ls <appid> -lA run/host/user-fonts`, what do you see?

`ls: cannot access 'run/host/user-fonts': No such file or directory`

**Comment by bbhtt** [comment]
Sorry it's `/run/host/user-fonts` missed the intial forward slash.

**Comment by mariinkys** [comment]
```
total 21964
-rw-rw-r--. 1 mariinkys mariinkys  5732824 May  3  2023 NotoSansJP-Regular.ttf
-rw-rw-r--. 1 mariinkys mariinkys  6192764 Aug 18  2023 NotoSansKR-Regular.ttf
-rw-rw-r--. 1 mariinkys mariinkys 10560380 Aug 18  2023 NotoSansSC-Regular.ttf
```

**Comment by bbhtt** [comment]
Then Flatpak is mounting them correctly inside the sandbox, beyond that something is wrong on your end. Not sure what.

But you can get a shell inside it with `--command=sh` and check if fonts are loaded with fc-match etc.

**Comment by mariinkys** [comment]
> Then Flatpak is mounting them correctly inside the sandbox, beyond that something is wrong on your end. Not sure what.
> 
> But you can get a shell inside it with `--command=sh` and check if fonts are loaded with fc-match/fc-list etc.

Thanks a lot for your help, I can't find any problem with the fonts tbh... But since it seems to work well on other systems I guess It's good enough for now...

**Comment by bbhtt** [comment]
> Should the screenshots show up on the store when looking at the app details?

https://docs.flathub.org/docs/for-app-authors/submission#i-dont-see-any-screenshots-from-the-test-builds-why

**Pull request closed without merging**